File: Data/Counter.pl
#!/bin/perl -w

# Count the numbers of Aleph Seq -records in file

use strict;
use utf8;
use List::MoreUtils qw(uniq);

binmode(STDOUT, ':utf8');

my @recordnumbers;
my $tiedosto = $ARGV[0];
my $linecount;

if( ! defined $tiedosto )
{
  die "Usage: perl Counter.pl inputfile\n";
}

open (my $inputfile, '<:utf8', $tiedosto);


while (<$inputfile>)
{
	my $id = substr($_, 0, 9); # Record id
	my $field = substr($_,10, 3); # Field code
	if ($field eq 'LOW')
	{
		push (@recordnumbers, $id);
	}
	$linecount++;	
}

my $totalnumberofrecords = uniq @recordnumbers;

print "The file $tiedosto contains $totalnumberofrecords records.\n";
close $inputfile;

File: counter.py
#!/usr/bin/env python
# -*- coding: utf-8 -*
# Reports the number of distinct Aleph Sequential -records in a file

import sys
import time
start_time = time.time()

if len(sys.argv) < 2:
	print("Usage: python counter.py inputfile")
	sys.exit()
else:
	tiedosto = sys.argv[1]

inputfile = open(tiedosto, "r")

recordCount = []
lineCount = 0
recordCounter = 0
firstline = inputfile.readline()
currentid = firstline[:8]
inputfile.seek(0)

for line in inputfile:
    id = line[:9]
    field = line[10:13]
    lineCount += 1
    if (id != currentid):
        currentid = id
        recordCount.append(id)
        recordCounter += 1
numberOfRecords = len(set(recordCount))
executionTime = (time.time() - start_time)
recordCounter += 1

print "The file " + tiedosto + " contains " + str(numberOfRecords) + " unique records (" + str(recordCounter) + " records in total) in " + str(lineCount) + " lines."
print "Processing took %.1f seconds." %  executionTime
inputfile.close()